sound/soc/sof/topology.c

```diff
@@ -582,8 +582,12 @@ static int sof_copy_tuples(struct snd_sof_dev *sdev, struct snd_soc_tplg_vendor_
 				found++;
 				(*num_copied_tuples)++;
 
-				/* stop if there's no space for any more new tuples */
-				if (*num_copied_tuples == tuples_size)
+				 /*
+				  * stop if there's no space for any more new tuples or if we've
+				  * reached the token_instance_num
+				  */
+				if (*num_copied_tuples == tuples_size ||
+				    found == num_tokens * token_instance_num)
```

Comment:
I would rather add a separate
```c
if (found == num_tokens * token_instance_num)
	return 0;
```
after the inner `for()` loop for better clarity.